packages/react-components/react-migration-v0-v9/src/components/Attachment/Attachment.tsx

```diff
@@ -0,0 +1,70 @@
+import { mergeClasses } from '@fluentui/react-components';
+import { useARIAButtonProps } from '@fluentui/react-aria';
+import * as React from 'react';
+
+import { useAttachmentBaseStyles, useAttachmentStyles } from './Attachment.styles';
+
+export const attachmentClassName = 'fui-Attachment';
+export const attachmentProgressContainerClassName = `${attachmentClassName}__progress-container`;
+export const attachmentProgressBarClassName = `${attachmentClassName}__progress`;
+
+export interface AttachmentProps extends React.HTMLAttributes<HTMLElement> {
+  actionable?: boolean;
+  disabled?: boolean;
+  progress?: string | number;
+  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+}
+
+export const Attachment = React.forwardRef<HTMLDivElement, AttachmentProps>((props, ref) => {
+  const { actionable, className, children, disabled, onClick, progress, onKeyDown, onKeyUp, ...rest } = props;
+  const attachmentBaseClass = useAttachmentBaseStyles();
+  const classes = useAttachmentStyles();
+
+  const handleClick = React.useCallback(e => {
+    if (disabled) {
+      e.preventDefault();
+      return;
+    }
+    if (onClick) {
+      onClick(e);
+    }
+  }, [onClick, disabled]);
+
+  const buttonProps = useARIAButtonProps('div', {
+    onClick: handleClick,

```

Comment:
```suggestion
  const buttonProps = useARIAButtonProps('div', {
    onClick: onClick,
    disabled,
```

`useARIAButtonProps` should be handlingt his for you just by passing the root props